library/core/src/iter/loop_desugar.rs

```diff
@@ -0,0 +1,76 @@
+use crate::iter::IntoIterator as RealIntoIterator;
+use crate::iter::TrustedRandomAccessNoCoerce;
+
+#[unstable(feature = "trusted_random_access", issue = "none")]
+#[doc(hidden)]
+
+pub trait IntoIterator {
```

Comment:
I think we should avoid a trait named `IntoIterator` that's not the actual trait -- it's really confusing, particularly for later PRs which'll touch just parts of this file and so won't see the RealIntoIterator import.